src/python/WMCore/ReqMgr/CherryPyThreads/StepChainParentageFixTask.py

```diff
@@ -12,24 +8,24 @@
 
 def getChildDatasetsForStepChainMissingParent(reqmgrDB, status):
     """
-    :param status: workflow status
-    :return: set of child dataset
+    Fetches data from ReqMgr2/CouchDB for workflows that need to
+    have their parentage information fixed.
+    :param status: string with the workflow status
+    :return: a flat list of dictionaries with request name and children datasets
     """
     results = reqmgrDB.getStepChainDatasetParentageByStatus(status)
-
-    requestsByChildDataset = defaultdict(set)
-
-    for reqName, info in viewitems(results):
-
-        for dsInfo in viewvalues(info):
-            for childDS in dsInfo["ChildDsets"]:
-                requestsByChildDataset[childDS].add(reqName)
-    return requestsByChildDataset
+    wflowChildrenDsets = []
+    for reqName, info in results.items():
+        childrenDsets = []
+        for _stepNum, dsetDict in info.items():
+            childrenDsets.extend(dsetDict['ChildDsets'])
+        wflowChildrenDsets.append({"workflowName": reqName, "childrenDsets": childrenDsets})
```

Comment:
Yes, I happen to use them interchangeably. I guess I will just keep it as is to avoid adding a mistake in the code.
I couldn't find the `ReqName` reference that you mention though.